.github/workflows/solidity-foundry-artifacts.yml

```diff
@@ -0,0 +1,479 @@
+            sol:
+              - 'contracts/src/v0.8/**/*.sol'
+              - *ignored   
+            product: &product
+              - 'contracts/src/v0.8/${{ inputs.product }}/**/*.sol'      
+              - *ignored
+
+      - name: Find modified changesets
+        uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
+        id: changes-dorny
+        with:
+          list-files: 'shell'
+          base: ${{ inputs.base_ref }}
+          filters: |
+            changeset:
+              - modified|added: 'contracts/.changeset/!(README)*.md'
+
+      # TODO remove when https://github.com/AurorNZ/paths-filter/issues/12 is fixed and we can use modifiers with it
+      - name: Get reference to compare to
+        id: get-base-ref
+        shell: bash
+        run: |
+          type=$(git cat-file -t "${{ inputs.base_ref }}")
+          if echo "$type" | grep -q commit; then
+              # we do not know to which branch the base ref belongs, so we fetch all branches
+              git fetch --all --prune
+              ref_to_use="${{ inputs.base_ref }}"
+          elif  echo "$type" | grep -q tag; then
+              git fetch origin ${{ inputs.base_ref }}
+              ref_to_use="origin/${{ inputs.base_ref }}" 
+          else
+              # must be branch
+              echo "::debug:: Base reference type: $type"
+              git fetch origin ${{ inputs.base_ref }}
+              ref_to_use="origin/${{ inputs.base_ref }}" 
+          fi
+          echo "ref_to_use=$ref_to_use" >> $GITHUB_OUTPUT
+
+      - name: Check if changes are not limited only to deleted files
+        if: ${{ steps.changes.outputs.sol == 'true' }}
+        shell: bash
+        run: |
+          echo "::debug::All modified contracts:"
+          echo ${{ steps.changes.outputs.sol_files }} | tr ',' '\n'                 
+          
+          IFS=','
+          for file in ${{ steps.changes.outputs.sol_files }}; do
+            # TODO remove when https://github.com/AurorNZ/paths-filter/issues/12 is fixed and we can use modifiers with it
+            status=$(git diff --name-status "${{ steps.get-base-ref.outputs.ref_to_use }}" HEAD -- "$file"  | awk '{ print $1 }')
+            if [ "$status" != "D" ]; then
+              exit 0
+            fi
+          done
+          
+          echo "::error:: All contract modifications are limited to deleted files. No artifacts will be generated."
+          echo "# Solidity Review Artifact NOT Generated" >> $GITHUB_STEP_SUMMARY
+          echo "Base Ref used: **${{ inputs.base_ref }}**" >> $GITHUB_STEP_SUMMARY
+          echo "Commit SHA used: **${{ github.sha }}**" >> $GITHUB_STEP_SUMMARY
+          echo "## Reason: Only deleted Solidity files found for ${{ inputs.product }}" >> $GITHUB_STEP_SUMMARY
+          
+          exit 1
+
+      - name: Check for changes outside of artifact scope
+        if: ${{ steps.changes.outputs.sol == 'true' }}
+        shell: bash
+        run: |
```

Comment:
I might consider moving the bash script into its own file for these longer scripts and then calling that script here.

Reply:
done

Reply:
I don't see a change for this?

Reply:
I put it in a dedicated action instead of bash script:
```
      - name: Check for changes outside of artifact scope
        uses: ./.github/actions/validate-artifact-scope
        if: ${{ steps.changes.outputs.sol == 'true' }}
        with:
          sol_files: ${{ steps.changes.outputs.sol_files }}
          product: ${{ inputs.product }}
```